aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -445,17 +443,18 @@ __global__ void nll_loss_backward_reduce_cuda_kernel_2d(
     bool size_average,
     int nframe,
     int ndim,
-    int n_classes,
+    int64_t n_classes,
     int64_t ignore_index) {
   const auto grad = -(size_average ? *grad_output / *total_weight
                                    : *grad_output);
 
   for (int i = threadIdx.x; i < nframe; i += NLL_LOSS_THREADS) {
-    int t = target[i];
-    if (t != static_cast<int>(ignore_index)) {
+    const int64_t t = static_cast<int64_t>(target[i]);
+    if (t != ignore_index) {
       CUDA_KERNEL_ASSERT(t >= 0 && t < n_classes);
-      grad_input[i * ndim + t] = weights != nullptr ? weights[t] * grad
-                                                    : grad;
+      const uint64_t index = static_cast<uint64_t>(i) * ndim + t;
```

Comment:
Let's use `int64_t`  or `uint64_t` to index `input` and `grad_input`, but let's be consistent.

Reply:
Also, in all the other kernels we have not cast `i`, so I don't think it's necessary to do so here either. You can simply do
```c++
const auto index = i * ndim + t;
```

Reply:
I think we typically use int64_t for indexing and name the integer types rather than auto in PyTorch code, but that is just my observation.